crates/ruff_python_ast/generate.py

```diff
@@ -0,0 +1,581 @@
+#!/usr/bin/python
+
+from __future__ import annotations
+from dataclasses import dataclass
+from pathlib import Path
+import re
+from subprocess import check_output
+import tomllib
+from typing import Optional
+
+
+def rustfmt(code: str) -> str:
+    return check_output(["rustfmt", "--emit=stdout"], input=code, text=True)
+
+
+def to_snake_case(node: str) -> str:
+    """Converts CamelCase to snake_case"""
+    return re.sub("([A-Z])", r"_\1", node).lower().lstrip("_")
+
+
+# ------------------------------------------------------------------------------
+# Read AST description
+
+root = check_output(["git", "rev-parse", "--show-toplevel"], text=True).strip()
+ast_path = Path(root).joinpath("crates", "ruff_python_ast", "ast.toml")
+out_path = Path(root).joinpath("crates", "ruff_python_ast", "src", "generated.rs")
+with ast_path.open("rb") as ast_file:
+    ast = tomllib.load(ast_file)
```

Comment:
I find the code a bit hard to read because it mixes top level statement with class and declarations. I'd prefer to instead have a `main` function that's called in `if __name__ == "__main__"`.

Reply:
Done